File: CSCI-3800-Advanced-Programming/HW03/Problem3_Java/src/Node.java
/*
    Patrick Tate
    CSCI 3800 HW03 Problem 3
 */

/*
    - Generic Node class to be used in conjunction with the CircularLinkedList class
    - Node contains two member variables, generic 'data' which can hold any data type,
    - and nextPtr of type 'Node<T>'
    - nextPtr points to the next Node in the linked list
 */
public class Node<T> {

    private T data; // generic data of type T
    private Node<T> nextPtr;

    // default constructor
    public Node(){
        this.data = null;
        this.nextPtr = null;
    }

    // getters
    public T getData() {
        return this.data;
    }
    public Node getNextPtr() {
        return this.nextPtr;
    }

    // setters
    public void setData(T data) {
        this.data = data;
    }
    public void setNextPtr(Node<T> nextPtr) {
        this.nextPtr = nextPtr;
    }
}

File: CSCI-3800-Advanced-Programming/HW03/Problem2_Python/node.py
# Patrick Tate
# CSCI 3800
# HW03 Problem 2

# Node class to be utilized in conjunction with the CircularLinkedList class
# Node has two member variables, 'data' which can store any data/values
# and 'nextPtr' which points to the next node in the class

class Node:

    # default initializer
    def __init__(self):
        self.__data = None
        self.__nextPtr = None # points to the next node in the linked list

    def setData(self, data):
        self.__data = data

    def setNextPtr(self, nextPtr):
        self.__nextPtr = nextPtr

    def getData(self):
        return self.__data

    def getNextPtr(self):
        return self.__nextPtr